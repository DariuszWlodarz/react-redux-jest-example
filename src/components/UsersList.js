import React from 'react';
import PropTypes from 'prop-types';
import User from './User';

const UsersList = ({users, onUserClickHandler}) => (
  <div>
    {users.map(user => (
      <User user={user} onClickHandler={onUserClickHandler} key={user.id} />
    ))}
  </div>
);

UsersList.propTypes = {
  users: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired
  })).isRequired,
  onUserClickHandler: PropTypes.func.isRequired
};

export default UsersList;