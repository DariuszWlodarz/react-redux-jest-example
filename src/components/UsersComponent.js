import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {loadUsersAction, deleteUserAction} from '../store/actions';
import Loader from './Loader';
import UsersList from './UsersList';

export const UsersComponent = ({isLoading = false, isLoaded = false, canLoad = false, users = [], loadUsersHandler, deleteUserHandler}) => (
  <div>
    <h1>USERS</h1>
    {isLoading && <Loader />}
    {users.length > 0 && <UsersList users={users} onUserClickHandler={deleteUserHandler} />}
    {canLoad && <button onClick={() => loadUsersHandler()}>load users</button>}
  </div>
);

UsersComponent.propTypes = {
  isLoading: PropTypes.bool,
  isLoaded: PropTypes.bool,
  canLoad: PropTypes.bool,
  users: PropTypes.array,
  loadUsersHandler: PropTypes.func.isRequired,
  deleteUserHandler: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  isLoading: state.loading,
  isLoaded: state.loaded,
  canLoad: !state.loading && state.data.length === 0,
  users: state.data
});

const mapDispatchToProps = {
  loadUsersHandler: loadUsersAction,
  deleteUserHandler: deleteUserAction
};

export default connect(mapStateToProps, mapDispatchToProps)(UsersComponent);