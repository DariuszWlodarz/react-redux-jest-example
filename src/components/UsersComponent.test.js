import React from 'react';
import {UsersComponent} from './UsersComponent';
import UsersList from './UsersList';
import Loader from './Loader';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

const users = [
  {
    id: 1,
    email: 'user1@example.com'
  },
  {
    id: 2,
    email: 'user2@example.com'
  }
];

test('snapshot', () => {
  const dummy = jest.fn();
  let component = shallow(<UsersComponent loadUsersHandler={dummy} deleteUserHandler={dummy} />);
  expect(toJson(component)).toMatchSnapshot();
  expect(component.find('button').length).toEqual(0);

  component = shallow(<UsersComponent canLoad={true} loadUsersHandler={dummy} deleteUserHandler={dummy} />);
  expect(toJson(component)).toMatchSnapshot();
  expect(component.find('button').length).toEqual(1);

  component = shallow(<UsersComponent isLoading={true} loadUsersHandler={dummy} deleteUserHandler={dummy} />);
  expect(toJson(component)).toMatchSnapshot();
  expect(component.find(Loader).length).toEqual(1);

  component = shallow(<UsersComponent loadUsersAction={dummy} loadUsersHandler={dummy} deleteUserHandler={dummy} isLoaded={true} users={users} />)
  expect(toJson(component)).toMatchSnapshot();
  expect(component.find(UsersList).length).toEqual(1);
  expect(component.find(Loader).length).toEqual(0);
  expect(component.find(UsersList).prop('users')).toEqual(users);
});