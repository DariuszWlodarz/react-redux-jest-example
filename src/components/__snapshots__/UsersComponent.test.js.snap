// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <h1>
    USERS
  </h1>
</div>
`;

exports[`snapshot 2`] = `
<div>
  <h1>
    USERS
  </h1>
  <button
    onClick={[Function]}
  >
    load users
  </button>
</div>
`;

exports[`snapshot 3`] = `
<div>
  <h1>
    USERS
  </h1>
  <Loader />
</div>
`;

exports[`snapshot 4`] = `
<div>
  <h1>
    USERS
  </h1>
  <UsersList
    onUserClickHandler={[Function]}
    users={
      Array [
        Object {
          "email": "user1@example.com",
          "id": 1,
        },
        Object {
          "email": "user2@example.com",
          "id": 2,
        },
      ]
    }
  />
</div>
`;
